backend/routes/flightStatus.js

```diff
@@ -81,21 +84,40 @@ try {
         orderBy: { scheduledDepartureDate: 'asc' },
     });
 
-    // Transform the data to match the TripCard component structure
-    const tripsData = upcomingTrips.map(trip => ({
+    const flightStatuses = await Promise.all(upcomingTrips.map(async (trip) => {
+        const response = await axios.get(`${API_BASE_URL}/flight-status`, {
```

Comment:
With this code, your endpoint might fail if there's an error fetching fresh data from amadeus, even if the error only happens for a single flight.

I'd suggest to:
1. Catch any errors so that your API doesn't fail if there's an exception fetching data from Amadeus.
2. Still show data from your own database if API fails, or perhaps ignore that single entry.

Up to you, but worth to consider how your product should handle those scenarios.

Reply:
That makes sense. I'll make the changes!